src/librustdoc/html/render/mod.rs

```diff
@@ -2532,6 +2532,16 @@ fn compare_impl<'a, 'b>(lhs: &'a &&Impl, rhs: &'b &&Impl) -> Ordering {
     compare_names(&lhs, &rhs)
 }
 
+fn compare_impl_polarity(a: &Impl, b: &Impl) -> Ordering {
+    match (a.inner_impl().polarity, b.inner_impl().polarity) {
+        (Some(ImplPolarity::Positive), Some(ImplPolarity::Negative)) => Ordering::Greater,
+        (Some(ImplPolarity::Negative), Some(ImplPolarity::Positive)) => Ordering::Less,
+        (Some(ImplPolarity::Positive), Some(ImplPolarity::Positive))
+        | (Some(ImplPolarity::Negative), Some(ImplPolarity::Negative)) => Ordering::Equal,
+        (None, _) | (_, None) => Ordering::Equal,
```

Comment:
What does it mean if the polarity is `None`? Is this the correct way to handle that case?

Reply:
```
$ rg polarity.*None src/librustdoc/
src/librustdoc/clean/blanket_impl.rs
136:                        polarity: None,

src/librustdoc/clean/auto_trait.rs
90:                        polarity = None;
```
I'm not sure what it means, but apparently it only happens for blanket impls and auto traits:
https://github.com/rust-lang/rust/blob/cb56a4420c8d41aab5c569c0fb6e51098d3a4745/src/librustdoc/clean/blanket_impl.rs#L136
https://github.com/rust-lang/rust/blob/cb56a4420c8d41aab5c569c0fb6e51098d3a4745/src/librustdoc/clean/auto_trait.rs#L89-L90

Reply:
I think this violates the transitivity requirement of `sort_by`.

Given `a = Positive; b = None; c = Negative` then

* a == b
* b == c
* a != c

Reply:
Then use the same logic used in `compare_impl`.

Reply:
I think this should be fixed on its own after https://github.com/rust-lang/rust/pull/80825/ - try rebasing.